app-tasks/rf/src/rf/ingest/io.py

```diff
@@ -0,0 +1,145 @@
+        add_overviews(merged_tif)
+        cog_path = convert_to_cog(merged_tif, local_dir)
+        updated_scene = upload_tif(cog_path, scene)
+        updated_scene.update()
+
+
+def add_overviews(tif_path):
+    logger.info('Adding overviews to %s', tif_path)
+    overviews_command = ['gdaladdo', tif_path]
+    subprocess.check_call(overviews_command)
+
+
+def convert_to_cog(tif_with_overviews_path, local_dir):
+    logger.info('Converting %s to a cog', tif_with_overviews_path)
+    out_path = os.path.join(local_dir, 'cog.tif')
+    cog_command = [
+        'gdal_translate',
+        tif_with_overviews_path,
+        # may need to add nodata, not sure yet
+        # '-a_nodata', '255'
+        '-co',
+        'TILED=YES',
+        '-co',
+        'COMPRESS=LZW',
+        '-co',
+        'COPY_SRC_OVERVIEWS=YES',
+        '-co',
+        'BIGTIFF=YES',
+        '-co',
+        'INTERLEAVE=BAND',
+        out_path
+    ]
+    subprocess.check_call(cog_command)
+    return out_path
+
+
+def fetch_imagery(image_locations, local_dir):
+    pool = Pool(cpu_count())
+    tupled = [(loc[0], loc[1], local_dir) for loc in image_locations]
+    try:
+        pool.map(fetch_imagery_uncurried, tupled)
+    finally:
+        pool.close()
+
+
+def fetch_imagery_uncurried(tup):
+    """Fetch imagery using a (location, filename, output_directory) tuple
+
+    This function exists so that something can be pickled and passed to pool.map,
+    since downloads are costly, functions are only pickleable if they're
+    defined at the top level of a module, and pool.map takes functions of one
+    argument.
+
+    Uncurried from:
+
+    uncurried :: (a -> b -> c) -> (a, b) -> c
+
+    in Haskell-land
+    """
+
+    fetch_image(tup[0], tup[1], tup[2])
+
+
+def fetch_image(location, filename, local_dir):
+    bucket, key = s3_bucket_and_key_from_url(location)
+    if bucket.startswith('landsat'):
+        extra_kwargs = {}
+    elif bucket.startswith('sentinel'):
+        extra_kwargs = {'RequestPayer': 'requester'}
+    # both sentinel and landsat have these uris in bucket.s3.amazonaws.com/...,
+    # so bucket and key from url does a bad job splitting correctly. follow up
+    # by splitting on '.' and taking the first one
+    bucket = bucket.split('.')[0]
+    logger.info('Fetching image from bucket %s with key %s', bucket, key)
+    dst = os.path.join(local_dir, filename)
+    with open(dst, 'w') as outf:
+        outf.write(
+            s3client.get_object(Bucket=bucket, Key=key,
+                                **extra_kwargs)['Body'].read())
+
+
+def merge_tifs(local_tif_paths, local_dir):
+    logger.info('Merging {} tif paths'.format(len(local_tif_paths)))
+    logger.debug('The files are:\n%s', '\n'.join(local_tif_paths))
+    merged_path = os.path.join(local_dir, 'merged.tif')
+    merge_command = [
+        'gdal_merge.py',
+        '-o',
+        merged_path,
+        # may need to add nodata, not sure yet
+        # '-a_nodata', '255',
```

Comment:
I think it's worth downloading and inspecting the tiffs to see if there is a correct nodata value to assign (similar to MODIS)